api/src/main/java/com/datatorrent/api/AppIdAwareStorageAgent.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.api;
+
+/**
+ * Interface to pass application id to storage agent
+ * 
+ * @since 3.2.0
+ *
+ */
+public interface AppIdAwareStorageAgent extends StorageAgent
+{
+  /**
+   * Save the application id so it can be used by StorageAgent for checkpointing
+   * 
+   * @param applicationId
+   */
+  public void saveAppId(String applicationId);
```

Comment:
Instead of just applicationId, can you pass Attributes. The DAG attributes has following members which may be useful for a storage agent.
- Application Name
- Application Id
- Application Path